.eslintrc.json

```diff
@@ -33,6 +33,19 @@
         "@typescript-eslint/no-empty-function": "off",
         "@typescript-eslint/explicit-function-return-type": "off",
         "@typescript-eslint/camelcase": "off",
+        "@typescript-eslint/naming-convention": [
+          "warn",
+          {
+            "selector": "variable",
+            "format": ["camelCase", "UPPER_CASE"],
```

Comment:
For interfaces, I believe we use the format:

```
MyInterface
```

Which wouldn't match the camel case or all caps regex?

Reply:
The `naming-convention` plugin is aware of ECMAScript IdentifierName contexts using the `selector` property. In this case, the format rule is only applied to Identifiers that are `variable`s. Other selectors exist such as `function`, `method`, `class`, `typeAlias`, and for the case you describe, `interface`. Therefore, this format will not apply to interface IdentifierNames such as `MyInterface`.

I'd be happy to add more `naming-convention` rules, especially since I am of the opinion that we should be a bit more strict. See https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#individual-selectors

I actually tested this against nodejs-bigquery and found only 3 linting problems related to this change, but only because changes were made to accommodate gts@2.